File: backend/app/schemas/hero_demo.py
from __future__ import annotations
from pydantic import BaseModel, Field
from typing import List, Optional, Dict, Any

class ReasoningTraceDTO(BaseModel):
    goal: str
    thought: str
    observation: str
    action_taken: str
    decision_rationale: str
    json_payload: Optional[Dict[str, Any]] = None

class HeroAuditLogDTO(BaseModel):
    id: str
    step_number: int
    actor: str  # Merchant, AI Embeddings Engine, Customer, Commerce Agent, Razorpay Gateway, Reconciliation Engine
    event_type: str
    description: str
    timestamp: str
    metadata: Dict[str, Any] = {}

class HeroRiskCheckDTO(BaseModel):
    risk_level: str  # LOW, MEDIUM, HIGH
    risk_score: int  # 0-100 (0 is best)
    fraud_flags: List[str] = []
    credit_limit_inr: float = 500000.0
    settlement_variance_inr: float = 0.0
    gst_compliance_status: str = "COMPLIANT"
    reconciliation_verified: bool = True

class HeroTransactionDTO(BaseModel):
    transaction_id: str
    order_id: str
    payment_id: Optional[str] = None
    gross_amount: float
    tax_amount: float
    discount_amount: float
    net_deposit: float
    gateway_fee: float
    gst_on_fee: float
    payment_method: str = "upi"
    status: str  # created, captured, reconciled
    journal_vouchers: List[Dict[str, Any]] = []
    timestamp: str

class HeroProductItemDTO(BaseModel):
    id: str
    sku: str
    name: str
    brand: str
    category: str
    price: float
    original_price: Optional[float] = None
    rating: float = 4.9
    image_url: str
    key_features: List[str] = []
    gst_rate_pct: float = 18.0
    active_offer: Optional[str] = None
    match_score_pct: int = 98

class HeroCartItemDTO(BaseModel):
    product_id: str
    sku: str
    name: str
    price: float
    quantity: int
    subtotal: float
    gst_rate_pct: float = 18.0
    image_url: str

class HeroMemoryProfileDTO(BaseModel):
    customer_id: str
    customer_name: str
    tier: str  # Enterprise Platinum, Growth Gold, Starter
    total_spend_inr: float
    orders_count: int
    aov_inr: float
    affinity_categories: List[str] = []
    risk_profile: str = "LOW RISK (Score: 4/100)"
    last_purchased_sku: str
    last_purchase_date: str
    loyalty_points: int

class HeroStepDataDTO(BaseModel):
    step_number: int
    step_key: str
    title: str
    subtitle: str
    actor: str
    status: str  # pending, in_progress, completed
    timestamp: str
    data: Dict[str, Any] = {}
    reasoning: ReasoningTraceDTO
    audit_log: HeroAuditLogDTO
    risk_check: HeroRiskCheckDTO
    transaction: Optional[HeroTransactionDTO] = None

class HeroScenarioDTO(BaseModel):
    id: str
    title: str
    business_type: str
    customer_name: str
    customer_email: str
    initial_prompt: str
    budget_inr: float
    target_category: str
    recommended_skus: List[str]
    upsell_sku: str
    future_skus: List[str]

class HeroDemoStateDTO(BaseModel):
    session_id: str
    scenario: HeroScenarioDTO
    current_step: int  # 1 to 10
    is_completed: bool = False
    steps: List[HeroStepDataDTO] = []
    active_cart_items: List[HeroCartItemDTO] = []
    cart_subtotal: float = 0.0
    cart_tax: float = 0.0
    cart_discount: float = 0.0
    cart_final: float = 0.0
    applied_coupon: Optional[str] = None
    order_id: Optional[str] = None
    payment_id: Optional[str] = None
    payment_link: Optional[str] = None
    reconciled: bool = False
    memory_profile: Optional[HeroMemoryProfileDTO] = None
    future_recommendations: List[HeroProductItemDTO] = []
    audit_logs: List[HeroAuditLogDTO] = []
    transactions: List[HeroTransactionDTO] = []

class StepExecutionRequestDTO(BaseModel):
    session_id: Optional[str] = None
    scenario_id: Optional[str] = "mumbai_retail_expansion"
    step_number: int = 1

class RunAllRequestDTO(BaseModel):
    session_id: Optional[str] = None
    scenario_id: Optional[str] = "mumbai_retail_expansion"
    delay_ms: int = 0

File: apps/customer-storefront/src/types/hero_demo.ts
export interface ReasoningTrace {
  goal: string;
  thought: string;
  observation: string;
  action_taken: string;
  decision_rationale: string;
  json_payload?: Record<string, any>;
}

export interface HeroAuditLog {
  id: string;
  step_number: number;
  actor: string;
  event_type: string;
  description: string;
  timestamp: string;
  metadata: Record<string, any>;
}

export interface HeroRiskCheck {
  risk_level: 'LOW' | 'MEDIUM' | 'HIGH';
  risk_score: number;
  fraud_flags: string[];
  credit_limit_inr: number;
  settlement_variance_inr: number;
  gst_compliance_status: string;
  reconciliation_verified: boolean;
}

export interface HeroJournalEntry {
  account: string;
  type: 'DEBIT' | 'CREDIT';
  amount: number;
  description: string;
}

export interface HeroTransaction {
  transaction_id: string;
  order_id: string;
  payment_id?: string;
  gross_amount: number;
  tax_amount: number;
  discount_amount: number;
  net_deposit: number;
  gateway_fee: number;
  gst_on_fee: number;
  payment_method: string;
  status: string;
  journal_vouchers: HeroJournalEntry[];
  timestamp: string;
}

export interface HeroProductItem {
  id: string;
  sku: string;
  name: string;
  brand: string;
  category: string;
  price: number;
  original_price?: number;
  rating: number;
  image_url: string;
  key_features: string[];
  gst_rate_pct: number;
  active_offer?: string;
  match_score_pct?: number;
}

export interface HeroCartItem {
  product_id: string;
  sku: string;
  name: string;
  price: number;
  quantity: number;
  subtotal: number;
  gst_rate_pct: number;
  image_url: string;
}

export interface HeroMemoryProfile {
  customer_id: string;
  customer_name: string;
  tier: string;
  total_spend_inr: number;
  orders_count: number;
  aov_inr: number;
  affinity_categories: string[];
  risk_profile: string;
  last_purchased_sku: string;
  last_purchase_date: string;
  loyalty_points: number;
}

export interface HeroStepData {
  step_number: number;
  step_key: string;
  title: string;
  subtitle: string;
  actor: string;
  status: 'pending' | 'in_progress' | 'completed';
  timestamp: string;
  data: Record<string, any>;
  reasoning: ReasoningTrace;
  audit_log: HeroAuditLog;
  risk_check: HeroRiskCheck;
  transaction?: HeroTransaction;
}

export interface HeroScenario {
  id: string;
  title: string;
  business_type: string;
  customer_name: string;
  customer_email: string;
  initial_prompt: string;
  budget_inr: number;
  target_category: string;
  recommended_skus: string[];
  upsell_sku: string;
  future_skus: string[];
}

export interface HeroDemoState {
  session_id: string;
  scenario: HeroScenario;
  current_step: number;
  is_completed: boolean;
  steps: HeroStepData[];
  active_cart_items: HeroCartItem[];
  cart_subtotal: number;
  cart_tax: number;
  cart_discount: number;
  cart_final: number;
  applied_coupon?: string;
  order_id?: string;
  payment_id?: string;
  payment_link?: string;
  reconciled: boolean;
  memory_profile?: HeroMemoryProfile;
  future_recommendations: HeroProductItem[];
  audit_logs: HeroAuditLog[];
  transactions: HeroTransaction[];
}
